core/provider/service_test.go

```diff
@@ -361,7 +361,9 @@ func (s *ServiceTestSuite) TestFetchResources() {
 	s.Run("should return error if got any from resource service", func() {
 		s.mockProviderRepository.EXPECT().Find(mock.MatchedBy(func(ctx context.Context) bool { return true })).Return(providers, nil).Once()
 		for _, p := range providers {
-			s.mockProvider.On("GetResources", mockCtx, p.Config).Return([]*domain.Resource{}, nil).Once()
+			s.mockProvider.On("GetResources", mockCtx, p.Config).Return([]*domain.Resource{
+				{ID: "test"},
```

Comment:
Previously, this test is error triggered on BulkUpsert, since we updated new changes, if GetResources returns empty value, it will skip BulkUpsert function which resulted no error